pkg/cache/server/options/options.go

```diff
@@ -74,26 +74,25 @@ func NewOptions(rootDir string) *Options {
 		EmbeddedEtcd:     *etcdoptions.NewOptions(rootDir),
 	}
 
-	o.ServerRunOptions.EnablePriorityAndFairness = false
 	o.SecureServing.ServerCert.CertDirectory = rootDir
 	o.SecureServing.BindPort = 6443
 	o.Etcd.StorageConfig.Transport.ServerList = []string{"embedded"}
-	// TODO: enable the watch cache, it was disabled because
-	//  - we need to pass a shard name so that the watch cache can calculate the key
-	//    we already do that for cluster names (stored in the obj)
-	//  - we need to modify wildcardClusterNameRegex and crdWildcardPartialMetadataClusterNameRegex
-	o.Etcd.EnableWatchCache = false
 	return o
 }
 
 func (o *Options) Complete() (*CompletedOptions, error) {
 	if servers := o.Etcd.StorageConfig.Transport.ServerList; len(servers) == 1 && servers[0] == "embedded" {
 		o.EmbeddedEtcd.Enabled = true
 	}
-
+	o.ServerRunOptions.EnablePriorityAndFairness = false
```

Comment:
why here in complete?

Reply:
because I need to overwrite the etcd options when the cache-server is run with the kcp-server and then turn off what needs to be turned off, for example:

```
cacheServerOptions := cacheoptions.NewOptions(s.Options.Extra.RootDirectory)

cacheServerEtcdOptions := *s.Options.GenericControlPlane.Etcd
cacheServerOptions.Etcd = &cacheServerEtcdOptions

completedCacheServerOptions, err := cacheServerOptions.Complete()
```

Reply:
have a look at https://github.com/kcp-dev/kcp/pull/1954/files#diff-e3fc4d4c96ce1c7a560285f676ceafdb8b2ec8c582113f1ffce8305a2b9bec37R39

Reply:
you cannot override non-pointers in Complete.